lua/telescope/pickers.lua

```diff
@@ -452,19 +454,43 @@ function Picker:find()
 
     local process_complete = function()
       -- TODO: We should either: always leave one result or make sure we actually clean up the results when nothing matches
+      local set_default_selection_index = function()
+        self:set_selection(self:get_row(self.default_selection_index))
+      end
+
       if selection_strategy == 'row' then
-        self:set_selection(self:get_selection_row())
+        if self._selection_row == nil and self.default_selection_index ~= nil then
+          set_default_selection_index()
+        else
+          self:set_selection(self:get_selection_row())
+        end
       elseif selection_strategy == 'follow' then
-        local index = self.manager:find_entry(self:get_selection())
-
-        if index then
-          local follow_row = self:get_row(index)
-          self:set_selection(follow_row)
+        -- TODO(conni2461): Currently does not work with default_selection_index
+        -- find_entry(self:get_selection()) will return always 1 because inside find_entry
+        -- we iterate over all entry_states and compare based on display, which is a function
+        -- and points to the same function for all entry_states. Which makes it impossible to
+        -- get another result other than 1.
+        -- I tested it briefly with find_files and it seams to be broken aswell.
+        -- Maybe a comparison based on ordinal rather than display would be better.
+        -- Will take another look at this
```

Comment:
@tjdevries I think `selection_strategy == 'follow'` is currently broken. I also tested this on current master with `find_files` and i think we can fix this if we would compare based on ordinal in `manager:find_entry`.